src/main/logic/Player.h

```diff
@@ -12,15 +12,28 @@
 
 namespace gs {
 
+enum class PlayerState {
```

Comment:
I am not sure if we should use player state or not. I am thinking that immunity after being spawn should be handled by the same future mechanic as power ups. So the only other states left are dead and alive, which in effect will act as visible or invisible. Maybe more discussion?